packager/certcache/certcache.go

```diff
@@ -440,3 +503,96 @@ func (this *CertCache) fetchOCSP(orig []byte, ocspUpdateAfter *time.Time) []byte
 	}
 	return respBytes
 }
+
+
+// Checks for cert updates every certCheckInterval hours. Never terminates.
+func (this *CertCache) maintainCerts(stop chan struct{}) {
+	// Only make one request per certCheckInterval, to minimize the impact
+	// on servers that are buckling under load.
+	ticker := time.NewTicker(CertCheckInterval)
+
+	for {
+		select {
+		case <-ticker.C:
+			this.updateCertIfNecessary()
+		case <-stop:
+			ticker.Stop()
+			return
+		}
+	}
+}
+
+// Returns true iff cert cache contains at least 1 cert.
+func (this *CertCache) hasCert() bool {
+	return len(this.certs) > 0 && this.certs[0] != nil
+}
+
+func (this *CertCache) getCert() *x509.Certificate {
+	if !this.hasCert() {
+		return nil
+	}
+	return this.certs[0]
+}
+
+// Set current cert with mutex protection.
+func (this *CertCache) setCerts(certs []*x509.Certificate) {
+	this.certsMu.Lock()
+	defer this.certsMu.Unlock()
+	this.certs = certs
+	this.certName = util.CertName(certs[0])
+}
+
+// Set new cert with mutex protection.
+func (this *CertCache) setNewCerts(certs []*x509.Certificate) {
+	this.renewedCertsMu.Lock()
+	defer this.renewedCertsMu.Unlock()
+	this.renewedCerts = certs
+}
+
+// Update the cert in the cache if necessary.
+func (this *CertCache) updateCertIfNecessary() {
+	log.Println("Updating cert if necessary");
+	if this.certFetcher == nil {
+		// Do nothing if certFetcher is not set.
+		log.Println("Certfetcher is not set, skipping cert updates.");
```

Comment:
This is not possible in this code flow.  All I know here is that certfetcher is not set since it's nil.  The reason is not known at this point (it was known during an earlier flow).  The logging will just be useful for debugging as an indication that certificate fetching is not happening.